aces/idt/generators/base_generator.py

```diff
@@ -576,6 +580,30 @@ def sort(self) -> None:
         self._samples_camera = self._samples_camera[indices]
         self._samples_reference = self._samples_reference[indices]
 
+        # Now we have sorted the samples lets remove any clipped samples
+        self.remove_clipping()
+
+    def remove_clipping(self):
+        """
+        Remove any clipping from the samples and references, if we detect any clipping
+        at the floor or the ceiling of the samples we remove them from both the samples
+        and the references.
+
+        This is done by taking 2 code values at 10-bit as a threshold, and checking if
+        any of the r, g, b values between samples are less than the threshold.
```

Comment:
I haven't tested and I could be wrong but this seems brittle, what about images with a lot of noise in the shadows?

Why are we also filtering the reference samples?

Reply:
2 parts

1) Handling Noise In The Shadows
We did discuss this, and discussed it in the working group and it did come up.
Myself and frankie were perhaps a bit optimistic going for the 2 code values at 10-bit, we discussed upping the threshold to 4 code values.

But you are totally correct

2) In The Sort we generate the reference samples at the same exposure value as the samples where captured
If we remove clipping from the samples, then we have to also remove the synthetically generated references too, otherwise we end up with two arrays. samples_camera and samples_reference which are of different shape.

So when these arrays pass through the rest of the code everything falls apart.

By removing the equivalent "clipped" reference" we keep the arrays with the same shape, and the references for those samples we removed are not needed as their counterparts have been removed.